internal/services/eventgrid/eventgrid_namespace_resource.go

```diff
@@ -0,0 +1,601 @@
+			}
+
+			if metadata.ResourceData.HasChange("topic_spaces_configuration") {
+				payload.Properties.TopicSpacesConfiguration = expandTopicSpacesConfigurationUpdate(model.TopicSpacesConfiguration)
+			}
+
+			if err = client.UpdateThenPoll(ctx, *id, payload); err != nil {
+				return fmt.Errorf("creating %s: %+v", *id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r EventGridNamespaceResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.NamespacesClient
+
+			id, err := namespaces.ParseNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := EventGridNamespaceResourceModel{
+				Name:          id.NamespaceName,
+				ResourceGroup: id.ResourceGroupName,
+			}
+
+			if model := existing.Model; model != nil {
+				state.Location = model.Location
+
+				if model.Sku != nil {
+					state.Sku = string(pointer.From(model.Sku.Name))
+					state.Capacity = pointer.From(model.Sku.Capacity)
+				}
+				flattenedIdentity, err := identity.FlattenSystemAndUserAssignedMapToModel(model.Identity)
+				if err != nil {
+					return fmt.Errorf("flattening `identity`: %+v", err)
+				}
+				state.Identity = pointer.From(flattenedIdentity)
+				state.Tags = pointer.From(model.Tags)
+
+				if props := model.Properties; props != nil {
+					state.ZoneRedundant = pointer.From(props.IsZoneRedundant)
+					state.TopicSpacesConfiguration = flattenTopicSpacesConfiguration(props.TopicSpacesConfiguration)
+					state.InboundIpRules = flattenInboundIPRules(props.InboundIPRules)
+					state.PublicNetworkAccess = string(pointer.From(props.PublicNetworkAccess))
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r EventGridNamespaceResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.NamespacesClient
+
+			id, err := namespaces.ParseNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r EventGridNamespaceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return namespaces.ValidateNamespaceID
+}
+
+func expandInboundIPRules(input []InboundIpRuleModel) *[]namespaces.InboundIPRule {
+	var ipRules []namespaces.InboundIPRule
+
+	if len(input) == 0 {
+		return &ipRules
```

Comment:
If we're not supplying any inbound IP rules I think we should be returning nil here instead of an empty InboundIPRule slice